pkg/codegen/hcl2/model/type_test.go

```diff
@@ -668,20 +668,32 @@ func TestUnifyType(t *testing.T) {
 }
 
 func TestRecursiveObjectType(t *testing.T) {
-	props := map[string]Type{
-		"data": NewOutputType(IntType),
+	makeType := func(prop string) Type {
```

Comment:
Nit - rename to `makeLinkedListType`?

Reply:
Should we consider adding some more tests for unions etc.?

Reply:
We could... I mostly extended the existing set of recursive tests to the new function. I don't think it makes the coverage lower but it's true it doesn't improve it either.